riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest.java

```diff
@@ -0,0 +1,264 @@
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.nike.riposte.server.componenttest.VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest.AfterSecurityRequestFilter.*;
+import static com.nike.riposte.server.componenttest.VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest.BeforeSecurityRequestFilter.*;
+import static java.util.Collections.singleton;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+@RunWith(DataProviderRunner.class)
+public class VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest {
+
+    private static Server server;
+    private static ServerConfig serverConfig;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        serverConfig = new RequestAndResponseFilterTestConfig();
+        server = new Server(serverConfig);
+        server.startup();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.shutdown();
+    }
+
+    @Test
+    public void shouldExecuteBeforeAndAfterRequestAndResponseFilters() {
+        ExtractableResponse response =
+                given()
+                        .baseUri("http://127.0.0.1")
+                        .port(serverConfig.endpointsPort())
+                        .basePath(BasicEndpoint.MATCHING_PATH)
+                        .header("force-security-error", "false")
+                        .log().all()
+                        .when()
+                        .get()
+                        .then()
+                        .log().headers()
+                        .extract();
+
+        assertThat(response.statusCode(), is(200));
+
+        assertThat(response.header(BEFORE_SECURITY_FILTER_RESPONSE_HEADER_KEY), is(BEFORE_SECURITY_FILTER_RESPONSE_HEADER_VALUE));
+        assertThat(response.header(AFTER_SECURITY_FILTER_RESPONSE_HEADER_KEY), is(AFTER_SECURITY_FILTER_RESPONSE_HEADER_VALUE));
+    }
+
+    @Test
+    public void shouldOnlyExecuteBeforeSecurityRequestFilterWhenSecurityErrorThrown() {
+        ExtractableResponse response =
+                given()
+                        .baseUri("http://127.0.0.1")
+                        .port(serverConfig.endpointsPort())
+                        .basePath(BasicEndpoint.MATCHING_PATH)
+                        .header("force-security-error", "true")
+                        .log().all()
+                        .when()
+                        .get()
+                        .then()
+                        .log().headers()
+                        .extract();
+
+        assertThat(response.statusCode(), is(401));
+
+        assertThat(response.header(BEFORE_SECURITY_FILTER_RESPONSE_HEADER_KEY), is(BEFORE_SECURITY_FILTER_RESPONSE_HEADER_VALUE));
```

Comment:
I'd recommend having four total response headers, two added by each of the filters:

```
before-security-filter-request-first-chunk-executed: [true/false]
before-security-filter-request-last-chunk-executed: [true/false]

after-security-filter-request-first-chunk-executed: [true/false]
after-security-filter-request-last-chunk-executed: [true/false]
```

Use the RequestInfo's attributes (not headers) to shuttle info to the `filterResponse(...)` side of the filter when the request-side filter method is executed - if the attribute exists then the method was executed, if not then it wasn't, letting you set the true/false value for the relevant response header appropriately. 

Then the assertions for the no-error-thrown test would look something like:
```
assertThat(response.header(BEFORE_SECURITY_FILTER_REQUEST_FIRST_CHUNK_EXECUTED)).isTrue();
assertThat(response.header(BEFORE_SECURITY_FILTER_REQUEST_LAST_CHUNK_EXECUTED)).isTrue();
assertThat(response.header(AFTER_SECURITY_FILTER_REQUEST_FIRST_CHUNK_EXECUTED)).isTrue();
assertThat(response.header(AFTER_SECURITY_FILTER_REQUEST_LAST_CHUNK_EXECUTED)).isTrue();
```
And the assertions for the error-thrown test would look like:
```
assertThat(response.header(BEFORE_SECURITY_FILTER_REQUEST_FIRST_CHUNK_EXECUTED)).isTrue();
assertThat(response.header(BEFORE_SECURITY_FILTER_REQUEST_LAST_CHUNK_EXECUTED)).isTrue();
assertThat(response.header(AFTER_SECURITY_FILTER_REQUEST_FIRST_CHUNK_EXECUTED)).isFalse();
assertThat(response.header(AFTER_SECURITY_FILTER_REQUEST_LAST_CHUNK_EXECUTED)).isFalse();
```

You don't need to use these exact names, but hopefully the intention is clear?